src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala

```diff
@@ -9,6 +9,10 @@ import scala.language.postfixOps
 sealed trait ExecutionStoreKey {
   def scope: Scope
   def index: Option[Int]
+  def tag: String = {
+    val shard = index.map(x => s":$x").getOrElse("")
```

Comment:
I actually did try to move this to the implicit class for IndexExecutionIndex and it surprisingly broke everything, and very silently.  I was kind of baffled... I tried it many ways: method as `toString`, `tag`.  It compiles fine, the IDE recognizes it as correct, but when I go to run a workflow, the same way I've been doing it for days now... I get this output:

```
[2015-10-22 08:07:37,517] [info] WorkflowManagerActor submitWorkflow input id = None, effective id = 6d88e5a5-f531-4d3a-b234-13e596d026e7
[2015-10-22 08:07:37,591] [info] WorkflowManagerActor Found no workflows to restart.
[2015-10-22 08:07:37,754] [info] SingleWorkflowRunnerActor: workflow ID 6d88e5a5-f531-4d3a-b234-13e596d026e7
[2015-10-22 08:07:37,764] [info] WorkflowActor [6d88e5a5] Start message received
[2015-10-22 08:07:37,922] [warn] SingleWorkflowRunnerActor: received unexpected message: CurrentState(Actor[akka://cromwell-system/user/WorkflowManagerActor/WorkflowActor-6d88e5a5-f531-4d3a-b234-13e596d026e7#1150537083],Submitted)
[2015-10-22 08:07:38,57] [info] SingleWorkflowRunnerActor: workflow finished with status 'Failed'.
[2015-10-22 08:07:38,58] [info] WorkflowActor [6d88e5a5] transitioning from Submitted to Failed.
```

Very mysterious... if you want we can sit down and try to figure it out but just so I can get things working I'm keeping it like this.